optaplanner-examples/src/test/java/org/optaplanner/examples/nqueens/solver/score/NQueensConstraintProviderTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.examples.nqueens.solver.score;
+
+import org.junit.Test;
+import org.optaplanner.examples.nqueens.domain.Column;
+import org.optaplanner.examples.nqueens.domain.NQueens;
+import org.optaplanner.examples.nqueens.domain.Queen;
+import org.optaplanner.examples.nqueens.domain.Row;
+import org.optaplanner.test.impl.score.stream.ConstraintVerifier;
+import org.optaplanner.test.impl.score.stream.SingleConstraintVerifier;
+
+public class NQueensConstraintProviderTest {
+
+    private final NQueensConstraintProvider constraintProvider = new NQueensConstraintProvider();
+    private final ConstraintVerifier<NQueens> constraintVerifier =
+            ConstraintVerifier.createFor(NQueens.class, Queen.class);
+
+    @Test
+    public void testSingleConstraint() {
+        Column column1 = new Column();
+        column1.setId(0L);
+        column1.setIndex(0);
+        Column column2 = new Column();
+        column2.setId(1L);
+        column2.setIndex(1);
+        Row row = new Row();
+        row.setId(0L);
+        row.setIndex(0);
+        Queen queen1 = new Queen();
+        queen1.setId(0L);
+        queen1.setRow(row);
+        queen1.setColumn(column1);
+        Queen queen2 = new Queen();
+        queen2.setId(1L);
+        queen2.setRow(row);
+        queen2.setColumn(column2);
```

Comment:
This mess is what I hoped to avoid by allowing the user to simply deserialize a ready-made solution.
Keep in mind that this is a trival example - testing rea-world constraints will become a nightmare, unless we allow working with ready-made solutions.

Reply:
I think this can be solved by using a helper method in the test or factory methods in the domain model. But, since methods have just a single return value, we'd have to wrap the set of objects in some kind of container (the solution, probably (obviously?)).

`<TLDR>`
**So, no matter if we get the solution (or any kind of container) with a factory/helper method or by unmarshalling it, `givenFacts()` needs to (also) accept that instead of varargs of objects.**
`</TLDR>`

On the other hand, Geoffrey's argument against that approach is also valid. Sometimes the objects can be created very easily and wrapping them into a solution is an unnecessary ceremony.